src/ACFFields.php

```diff
@@ -0,0 +1,138 @@
+ *
+ * @since 0.6
+ */
+class ACFFields extends AbstractSteppable {
+	const ACFML_KEY = 'wpml_cf_preferences';
+	const TYPES     = [
+		0 => 'none', // WPML_IGNORE_CUSTOM_FIELD.
+		1 => 'copy', // WPML_COPY_CUSTOM_FIELD.
+		2 => 'translate', // WPML_TRANSLATE_CUSTOM_FIELD.
+		3 => 'copy_once', // WPML_COPY_ONCE_CUSTOM_FIELD.
+	];
+
+	/**
+	 * Returns the action name.
+	 *
+	 * @since 0.6
+	 *
+	 * @return string
+	 */
+	public function getName() {
+		return 'process_acf_fields';
+	}
+
+	/**
+	 * Returns the processing message.
+	 *
+	 * @since 0.6
+	 *
+	 * @return string
+	 */
+	protected function getMessage() {
+		return esc_html__( 'Processing ACF fields', 'wpml-to-polylang' );
+	}
+
+	/**
+	 * Processes the ACF fields.
+	 *
+	 * @since 0.6
+	 *
+	 * @return void
+	 */
+	protected function handle() {
+		global $wpdb;
+
+		$batch_size = absint( WPML_TO_POLYLANG_QUERY_BATCH_SIZE / 100 ); // 50 by default, to limit the size of the UPDATE query.
+		$offset     = absint( ( $this->step * $batch_size ) - $batch_size );
+		$results    = $wpdb->get_results(
+			$wpdb->prepare(
+				"SELECT ID, post_content
+				FROM {$wpdb->posts}
+				WHERE post_type = 'acf-field' AND post_content LIKE %s
+				ORDER BY ID ASC
+				LIMIT %d, %d",
+				'%"' . $wpdb->esc_like( self::ACFML_KEY ) . '"%',
+				$offset,
+				$batch_size
+			)
+		);
+
+		$values = [];
+
+		foreach ( $results as $field ) {
+			$post_content = maybe_unserialize( $field->post_content );
+
+			if ( ! is_array( $post_content ) ) {
+				// Unserialization failure.
+				continue;
+			}
+
+			if ( ! is_numeric( $post_content[ self::ACFML_KEY ] ) || ! isset( self::TYPES[ $post_content[ self::ACFML_KEY ] ] ) ) {
+				// Should not happen.
+				$post_content['translations'] = 'none';
+			} else {
+				$post_content['translations'] = self::TYPES[ $post_content[ self::ACFML_KEY ] ];
+			}
+
+			$post_content = serialize( $post_content ); // PHPCS:ignore WordPress.PHP.DiscouragedPHPFunctions.serialize_serialize
+
+			if ( ! is_string( $post_content ) ) {
+				// Serialization failure.
+				continue;
+			}
+
+			$values[ $field->ID ] = sprintf( "WHEN %d THEN '%s'", $field->ID, addcslashes( esc_sql( $post_content ), "'" ) );
+		} // End foreach.
+
+		if ( empty( $values ) ) {
+			return;
+		}
+
+		// Update the whole batch in one query.
+		$wpdb->query(
+			sprintf(
+				"UPDATE {$wpdb->posts}
+				SET post_content = CASE ID
+					%s
+					ELSE post_content
+				END
+				WHERE ID IN (%s)",
+				implode( "\n", $values ), // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+				implode( "\n", array_keys( $values ) ) // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+			)
+		);
+	}
+
+	/**
+	 * Returns the number of ACF fields to update.
+	 *
+	 * @since 0.6
+	 *
+	 * @return int
+	 */
+	protected function getTotal() {
+		global $wpdb;
+
+		return (int) $wpdb->get_var(
+			$wpdb->prepare(
+				"SELECT COUNT(*)
+				FROM {$wpdb->posts}
+				WHERE post_type = 'acf-field' AND post_content LIKE %s",
```

Comment:
```suggestion
				WHERE post_type = 'acf-field' AND CONTAINS (post_content, %s)",
```
`CONTAINS` *can* be faster.

Reply:
1. `CONTAINS` doesn't exist in mysql, but `INSTR` does the same thing.
2. From what I've read, `CONTAINS` and `INSTR` do a fuzzy match, while we want an exact match.
3. Note: `CONTAINS` can be faster, but only if the column has a full text index. `post_content` seems to have it but I'm not 100% sure, I'm not familiar with this concept.

Reply:
I thought `CONTAINS` was available for MySQL. Maybe it's an alias for `INSTR` (need to be checked though).
But regarding fuzzy search, I don't think so. Especially if you don't need pattern (like here with `self::ACFML_KEY`).

Reply:
I'll let @Chouby chose what he prefers.

Reply:
That's what I've read in several places. However, [the doc](https://learn.microsoft.com/en-us/sql/t-sql/queries/contains-transact-sql?view=sql-server-ver16) says it can be used for fuzzy or precise search.

Reply:
From your source: "Searches for precise or fuzzy (less precise) matches to single words and phrases". Meaning it can do both. So in order to perform fuzzy search you have to provide `contains_search_condition` according to the doc. Also, careful with Microsoft documentation because SQL Server and Azure SQL aren't exactly the same as MySQL (or Maria DB for instance).